lib/configproxy.js

```diff
@@ -337,6 +340,7 @@ ConfigurableProxy.prototype.handle_proxy = function (kind, req, res) {
         this.handle_proxy_error(404, kind, req, res);
         return;
     }
+    this.emit("proxy_request", req, res);
```

Comment:
Care must be taken with how users work with this event, since it can affect all listeners and the proxy itself.